misc/json.c

```diff
@@ -293,6 +294,7 @@ static int json_append(bstr *b, const struct mpv_node *src, int indent)
     case MPV_FORMAT_NONE:
         APPEND(b, "null");
         return 0;
+    case MPV_FORMAT_BOOL:
     case MPV_FORMAT_FLAG:
         APPEND(b, src->u.flag ? "true" : "false");
```

Comment:
When it's `MPV_FORMAT_BOOL` then the union is actually a `u.bool_` right?
We can't read a bool as an int (and vice versa), that will lead to problems as sfan5 already showed with his python script. Specifically if e.g. the bool is e.g. 1 byte and the int is 4 bytes, then reading `u.bool_` as `u.flag` will also read 3 bytes of undefined value.

Reply:
Yeah that's correct. `json_write` (which calls this) is basically the equivalent of `set_property` (I think). So it should be a separate case.